src/components/VideoPlayer/VideoPlayerControls/ProgressBar/index.js

```diff
@@ -0,0 +1,69 @@
+import PropTypes from 'prop-types';
+import React, {useEffect, useState} from 'react';
+import {Gesture, GestureDetector} from 'react-native-gesture-handler';
+import Animated, {runOnJS, useAnimatedStyle, useSharedValue} from 'react-native-reanimated';
+import {usePlaybackContext} from '@components/VideoPlayerContexts/PlaybackContext';
+import useThemeStyles from '@hooks/useThemeStyles';
+
+const propTypes = {
+    duration: PropTypes.number.isRequired,
+
+    position: PropTypes.number.isRequired,
+
+    seekPosition: PropTypes.func.isRequired,
+
+    togglePlayCurrentVideo: PropTypes.func.isRequired,
+};
+
+const defaultProps = {};
+
+function ProgressBar({togglePlayCurrentVideo, duration, position, seekPosition}) {
+    const styles = useThemeStyles();
+    const {pauseVideo} = usePlaybackContext();
+    const [sliderWidth, setSliderWidth] = useState(1);
+    const progressWidth = useSharedValue(0);
+
+    const onSliderLayout = (e) => {
+        setSliderWidth(e.nativeEvent.layout.width);
+    };
+
+    const pan = Gesture.Pan()
+        .onBegin(() => {
+            runOnJS(pauseVideo)();
```

Comment:
Why do we need to pause the video here on start? One thing I notice is that if you just tap one position of the progress bar, it doesn't immediately goes to that position until you "pan" to run the `onChange()` function below, which feels unusual to me, especially compared with how you can just tap on a position on the progress bar with YT and Vimeo's players.

https://github.com/Expensify/App/assets/2229301/f7e658e9-5d0d-4e1c-add3-e0e00aa0d6cb

Reply:
That behaviour was caused by this we only handled pan gesture. I've just added tap and long press gesture so it will feel more natural as in other players.